scout/server/blueprints/cases/controllers.py

```diff
@@ -652,6 +653,16 @@ def phenotypes_genes(store, case_obj):
             "genes": ", ".join(sorted(gene_list)),
         }
 
+    if case_obj.get("dynamic_gene_list"):
+        gene_list = [
+            gene.get("hgnc_symbol") or str(gene["hgnc_id"])
+            for gene in case_obj["dynamic_gene_list"]
+        ]
+        hpo_genes["Case dynamic gene list"] = {
+            "description": "custom",
+            "genes": ", ".join(sorted(gene_list)),
```

Comment:
Actually what you really would want is sets and subtract the genes that came from `dynamic_panel_phenotypes` right? The difference would be either `custom` adds or stuff no longer concurrent due to later day changes in the HPO-to-gene mapping. Yiekes, that is actually scary as well: let's absolutely not print any gene to that document that has not really been on the panel for analysis. 😊

Reply:
> let's absolutely not print any gene to that document that has not really been on the panel for analysis

But then this check should also be done for the genes coming from HPO terms, right?

Reply:
> Actually what you really would want is sets and subtract the genes that came from `dynamic_panel_phenotypes` right?

Right! I didn't notice that when you add genes from a phenotypes they get also added in dynamic_gene_list. I'll fix, thanks!

Reply:
Ok this is a bit more complicated than I thought. For instance imagine you have have an HPO panel composed of genes associated to one or more HPO terms. If you export it then all good. 

But if you add a custom gene, then `dynamic_panel_phenotypes` field becomes empty, and that's why it didn't print anything in the first place. So there is really not much to subtract here, because only the dynamic_gene_list is populated.. What I'd to fix it then is checking if dynamic_panel_phenotypes is empty, and if it is print the list of genes in dynamic_gene_list only. But let me know if you have a better suggestion..!

Reply:
Mmm the more I think about this and the more it looks like a bug. dynamic_panel_phenotypes shouldn't be erased when you add a non-hpo custom gene. I'll try to fix it and propose a solution on this PR

Reply:
Ok, that sounds well in keeping with previous mode of operation then! 
I very much like that, and I would like it even better if it had turned out we do the same thing (clear the intermediate dynamic HPO term list) every time the dynamic gene list was generated.

I will right back say before I read you PR, I would not have remembered we stored the `dynamic_panel_phenotypes` for use beyond repopulating the `dynamic_gene_list`. The two should really be kept well in sync with how they are presented on the interface, otherwise we can end up in version issues with the HPO-gene mapping. Assume for instance a user Alice does some filtering with a set of HPO terms, generating a dynamic panel and using it. Then for some reason they wait a bit with generating the gene list pdf for the report - e.g. while waiting for a doctor Chiara to respond to a comment about one of the genes. In the meantime Bob at CG does a monthly (cough) update of HPO terms. Alice may now generate a pdf with genes that were actually not checked, and doctor Chiara sees their new favourite gene checked, while in fact it wasn't.

Reply:
Yes I see your point! So it isn't a bug but a way to ensure that the HPO genes are up to date with the analysis. I'll revert!

Reply:
Good stuff! Yes, or perhaps rather out-of-date, but completely in line with the analysis.

On the account of reading-before-ranting-next-time, L645 actually kills most of my second part worries by checking that filtered out (or as it were, novel) genes are not shown anyway. So all good now methinks?

Reply:
I think so! I've added a comment to make it more easy to remember next time we touch this code